### PR TITLE
feat: Upstream rules from EMR and webapp

### PR DESCRIPTION
**Pull request by choochootrain**
BREAKING CHANGE: Upgrade to eslint v5

### CODE REVIEW

File: react.json [hunk 1]
```diff
@@ -31,31 +31,32 @@
...
     "no-await-in-loop": 0,
+    "no-param-reassign": 0,
+    "no-confusing-arrow": 0,
+    "no-nested-ternary": 0,
     "camelcase": 0,
     "arrow-parens": [
       "error",
       "as-needed"
     ],
-    "no-confusing-arrow": [
```
Comment (choochootrain) [inline react.json:39]: enabling `allowParens` conflicts with prettier, which would try to remove the extra parens.

this is basically idiomatic in react code so i dont think its a big deal to disable

File: react.json [hunk 2]
```diff
@@ -31,31 +31,32 @@
...
     "import/extensions": 0,
     "import/no-unresolved": 0,
     "react/jsx-filename-extension": 0,
     "react/require-default-props": 0,
     "react/sort-comp": 0,
     "react/prop-types": 0,
     "react/jsx-closing-bracket-location": 0,
+    "react/forbid-prop-types": 0,
     "jsx-a11y/anchor-is-valid": 0,
+    "jsx-a11y/click-events-have-key-events": 0,
```
Comment (choochootrain) [inline react.json:57]: i'll create an a11y audit task to address these issues (and the existing a11y issues that are `eslint-disabled`)

**Comment by kylealwyn** [comment]
Are you going to drop in the generated `codeclimate.json` to use in prepare step in this PR or separate?

File: react.json (hunk 2 above)
Comment (choochootrain) [reply react.json:57]: created

**Comment by choochootrain** [comment]
i'll first test with https://raw.githubusercontent.com/sappira-inc/eslint-config-sappira/feat/upstream-rules/react.json since codeclimate does support prettier and airbnb eslint configs as special cases and that's all we depend on here.

if thats working we can merge in and ill revert `.codeclimate.yml` otherwise i'll drop the squashed eslint config into this PR

**Comment by choochootrain** [comment]
@kylealwyn codeclimate checks are passing now, once this is published i'll update dependencies and revert `prepare` to pull from `unpkg`

**Comment by choochootrain** [comment]
codeclimate is appeased again

**Review by kylealwyn — APPROVED** [review]

**Pull request merged**